src/BugsnagUnity/Native/Fallback/CacheManager.cs

```diff
@@ -124,41 +132,56 @@ private void CheckForMaxCachedPayloads(string[] payloads, int maxPayloads)
 
         private void RemoveOldestFiles(string[] filePaths, int numToRemove)
         {
-            var ordered = filePaths.OrderBy(file => File.GetCreationTimeUtc(file)).ToArray();
-            foreach (var file in ordered.Take(numToRemove))
+            try
             {
-                File.Delete(file);
+                var ordered = filePaths.OrderBy(file => File.GetCreationTimeUtc(file)).ToArray();
+                foreach (var file in ordered.Take(numToRemove))
+                {
+                    File.Delete(file);
```

Comment:
If one delete fails then it will bail out of the loop and not try any others - is that what you want?  I might have thought that wrapping the try/catch only around the `Delete` line itself would be better?  (Same for the other methods of course).

Reply:
fixed 👍

Reply:
And the other methods in the file?